scripts/ci/payload-generator/main.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	fp "path/filepath"
+
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/environment"
+	"github.com/ActiveState/cli/internal/exeutils"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/installation"
+)
+
+var (
+	log = func(msg string, vals ...any) {
+		fmt.Fprintf(os.Stdout, msg, vals...)
+		fmt.Fprintf(os.Stdout, "\n")
+	}
+	logErr = func(msg string, vals ...any) {
+		fmt.Fprintf(os.Stderr, msg, vals...)
+		fmt.Fprintf(os.Stderr, "\n")
+	}
+)
+
+// The payload-generator is an intentionally very dumb runner that just copies some files around.
+// This could just be a bash script if not for the fact that bash can't be linked with our type system.
+func main() {
+	if err := run(); err != nil {
+		logErr("%s", err)
+		os.Exit(1)
+	}
+}
+
+func run() error {
+	root := environment.GetRootPathUnsafe()
+	buildDir := fp.Join(root, "build")
+	payloadDir := fp.Join(buildDir, "payload")
+
+	return generatePayload(buildDir, payloadDir)
+}
+
+func generatePayload(buildDir, payloadDir string) error {
+	emsg := "generate payload: %w"
+
+	payloadBinDir := fp.Join(payloadDir, "bin")
+
+	if err := fileutils.MkdirUnlessExists(payloadBinDir); err != nil {
+		return fmt.Errorf(emsg, err)
+	}
+
+	installDirMarker := installation.InstallDirMarker
+	log("Creating install dir marker in %s", payloadDir)
+	if err := fileutils.Touch(fp.Join(payloadDir, installDirMarker)); err != nil {
+		return fmt.Errorf(emsg, err)
+	}
```

Comment:
This isn't incorporating the changes I pinged you about:

https://github.com/ActiveState/cli/commit/52f88ec1bb7c28bb095682e8c03a150cd2b3c812#diff-44cb367e4317af551646a2ed53fb338e5addea2dfc547b6cea02c6eb1874b3c5